### ISSUE TITLE
Docker Version 4.34.2 (167172) Completely Bricked

### ISSUE THREAD

**Issue by sidpremkumar** [docker/for-mac#7431]
### Description

```
running engine: waiting for the Docker API: engine linux/virtualization-framework failed to run: running VM: com.docker.virtualization: process terminated unexpectedly: accepting AF_VSOCK connection: AF_VSOCK listener closed
```

Can't seem to start or build any containers. 

Huge issue for us as we're stuck unable to push anything to production. 

### Reproduce

Attempt to start docker, and build an image. 

It will hang, in another terminal window try `docker container ls` and you'll get
```
request returned Internal Server Error for API route and version http://%2FUsers%2Fsidpremkumar%2F.docker%2Frun%2Fdocker.sock/v1.47/containers/json, check if the server supports the requested API version
```

And the server is dead. 

### Expected behavior

It should be able to start

### docker version

```bash
Client:
 Version:           27.2.0
 API version:       1.47
 Go version:        go1.21.13
 Git commit:        3ab4256
 Built:             Tue Aug 27 14:14:45 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux
Cannot connect to the Docker daemon at unix:///Users/sidpremkumar/.docker/run/docker.sock. Is the docker daemon running?

(can't get server info since it can't even start)
```

### docker info

```bash
Client:
 Version:    27.2.0
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.2-desktop.1
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.2-desktop.2
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.34
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-debug
  desktop: Docker Desktop commands (Alpha) (Docker Inc.)
    Version:  v0.0.15
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-desktop
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.25
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.3.0
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.13.0
    Path:     /Users/sidpremkumar/.docker/cli-plugins/docker-scout

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 6
 Server Version: 27.2.0
 Storage Driver: overlayfs
  driver-type: io.containerd.snapshotter.v1
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fc6bcff51318944179630522a095cc9dbf9f353
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.10.4-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 10
 Total Memory: 7.655GiB
 Name: docker-desktop
 ID: 45eeed66-1e64-4f0e-b019-95ff5834ade8
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=unix:///Users/sidpremkumar/Library/Containers/com.docker.docker/Data/docker-cli.sock
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

4521A47D-0E1D-4E3B-9888-085950F7DC00/20240925022427

### Additional Info

This is a huge issue for us personally and I imagine others. We are sitting suck unable to push anything to prod. 

Any help here is greatly appreciated

**Comment by sidpremkumar** [comment]
Restarting the computer seems to temporarily help, but not a long term solution as the issue comes back eventually.

**Comment by sidpremkumar** [comment]
Closing this as restarting the computer a couple of times now and its seemed to have gone away for now.

**Comment by dkbhadeshiya** [comment]
Hi, 

Same issue occurred with me as well. 
I tried docker buildx command to build an image for another platform architecture and it bricked whole installation. 

It can be reproduced with version 4.34.3 (170107). 

I tried reinstalling and again got bricked with `docker buildx build --platform=linux/amd64`

Information: 
Version: 4.34.3 (170107)
Engine: 27.2.0
Compose: v2.29.2-desktop.2
Credential Helper: v0.8.2
Kubernetes: v1.30.2
MacOS: 15.0.1 (24A348)
